packages/vite/src/node/server/index.ts

```diff
@@ -269,7 +273,10 @@ export async function createServer(
   const httpServer = middlewareMode
     ? null
     : await resolveHttpServer(serverConfig, middlewares)
-  const ws = createWebSocketServer(httpServer, config)
+  const ws =
+    config.server.hmr === false
+      ? createMockWebSocketServer()
```

Comment:
**suggestion (if-minor):** We may pass at least `config` to `createMockWebSocketServer`
That way we are able to use its `logger` in line `95` and `99` and don't need to rely on `console.info`

Also we may to color it with `chalk`
You can see an example for that in line `49`

_All line numbers are related to `packages/vite/src/node/server/ws.ts`_

Reply:
Excellent point!